apps/web/pages/settings/my-account/general.tsx

```diff
@@ -45,42 +45,51 @@ interface GeneralViewProps {
   user: RouterOutputs["viewer"]["me"];
 }
 
-const WithQuery = withQuery(trpc.viewer.public.i18n, undefined, { trpc: { context: { skipBatch: true } } });
-
 const GeneralQueryView = () => {
   const { t } = useLocale();
 
+  const { data: locale, isLoading: localeLoading } = trpc.viewer.public.locale.useQuery();
   const { data: user, isLoading } = trpc.viewer.me.useQuery();
-  if (isLoading) return <SkeletonLoader title={t("general")} description={t("general_description")} />;
+  if (isLoading || localeLoading)
+    return <SkeletonLoader title={t("general")} description={t("general_description")} />;
   if (!user) {
     throw new Error(t("something_went_wrong"));
   }
-  return (
-    <WithQuery
-      success={({ data }) => <GeneralView user={user} localeProp={data.locale} />}
-      customLoader={<SkeletonLoader title={t("general")} description={t("general_description")} />}
-    />
-  );
+  return <GeneralView user={user} localeProp={locale?.locale ?? "en"} />;
 };
 
 const GeneralView = ({ localeProp, user }: GeneralViewProps) => {
   const router = useRouter();
   const utils = trpc.useContext();
   const { t } = useLocale();
+  const queryClient = useQueryClient();
 
   const mutation = trpc.viewer.updateProfile.useMutation({
-    onSuccess: async () => {
-      // Invalidate our previous i18n cache
-      await utils.viewer.public.i18n.invalidate();
+    onSuccess: async (value) => {
+      const previousLocale = utils.viewer.public.locale.getData();
+
+      if (value?.locale) {
+        const locale = value?.locale;
+
+        await utils.viewer.public.i18n.fetch({ locale });
```

Comment:
fetch the new locale

Reply:
Shouldn't we simply be able to invalidate both i18n and locale? Even though they have infinite staleTime, manually invalidating should work still.